### PR TITLE
Promote OS probe results to DiskStorageService

### PR DESCRIPTION
**Pull request by jpnurmi**
In addition to the "Installation type" page, the same information about existing OS installations is also needed on the upcoming "Install X alongside Y" page. We can share the logic via DiskStorageService.

Ref: #819

### CODE REVIEW

File: packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_model.dart [hunk 1]
```diff
@@ -74,22 +73,10 @@ class InstallationTypeModel extends SafeChangeNotifier {
...
-                .whereType<OsProber>(),
-          )
-          .toList();
-      notifyListeners();
-    });
-  }
+  List<OsProber>? get existingOS => _diskService.existingOS;
+
+  /// Initializes the model.
+  Future<void> init() async {}
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/pages/installation_type/installation_type_model.dart:79]: i'm leaving an empty method here because soon there will be initialization steps again :)

**Review by CarlosNihelton — APPROVED** [review]
It's quite hard to follow the dance of the new partitioning API, but the end result is becoming really nice!
